pkg/validate/validate_values_test.go

```diff
@@ -129,6 +132,52 @@ global:
 	}
 }
 
+func TestValidateValuesFromProfile(t *testing.T) {
+	tests := []struct {
+		desc     string
+		profile  string
+		wantErrs util.Errors
+	}{
+		{
+			desc:    "default profile",
+			profile: "default",
+		},
+		{
+			desc:    "demo profile",
+			profile: "demo",
+		},
+		{
+			desc:    "demo-auth profile",
+			profile: "demo-auth",
+		},
+		{
+			desc:    "minimal profile",
+			profile: "minimal",
+		},
+		{
+			desc:    "sds profile",
+			profile: "sds",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			pf, err := helm.ReadProfileYAML(tt.profile)
+			if err != nil {
+				t.Fatalf("fail to read profile: %s", tt.profile)
+			}
+			val, _, err := manifest.ParseK8SYAMLToIstioControlPlaneSpec(pf)
+			if err != nil {
+				t.Fatalf(" fail to parse profile to ISCP: (%s), got error %s", tt.profile, err)
+			}
+			errs := CheckValues(val.Values)
+			if gotErr, wantErr := errs, tt.wantErrs; !util.EqualErrors(gotErr, wantErr) {
+				t.Errorf("CheckValues of (%v): gotErr:%s, wantErr:%s", tt.profile, gotErr, wantErr)
+			}
+		})
+	}
+
+}
+
```

Comment:
would be great to add another unit test TestValidateValuesFailure which would test a couple of bad profiles to ensure we're getting expected errors. nothing fancy, it could be just some inline yaml.

Reply:
actually there is existing test for bad field of .values part, is still necessary creating extra tests for bad profiles if they are provided as simple inline yaml as well(which would be just adding values as root)